migrations/upgrades/20191128055729_update_directus_files.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+
+use Phinx\Migration\AbstractMigration;
+use function \Directus\get_random_string;
+
+class UpdateDirectusFiles extends AbstractMigration
+{
+    public function change()
+    {
+        $fieldsTable = $this->table('directus_fields');
+        $filesTable = $this->table('directus_files');
+
+        if($this->checkFieldExist('directus_files', 'filename')){
+            $this->execute(\Directus\phinx_update(
+                $this->getAdapter(),
+                'directus_fields',
+                [
+                    'field' => 'filename_disk'
+                ],
+                ['collection' => 'directus_files', 'field' => 'filename']
+            ));
+        }
+
+        if (!$filesTable->hasColumn('filename_disk')) {
+            $filesTable->renameColumn('filename', 'filename_disk');
+        }
+
+        if (!$filesTable->hasColumn('filename_download')) {
+            $filesTable->addColumn('filename_download', 'string', [
+                'limit' => 255,
+                'null' => false
+            ])->save();
+        }
+        if(!$this->checkFieldExist('directus_files', 'filename_download')){
+            $fieldsTable->insert([
+                'collection' => 'directus_files',
+                'field' => 'filename_download',
+                'type' => 'string',
+                'interface'=>'text-input',
+                'locked' => 1
+            ])->save();
+        }
+
+        // adds private_hash in directus_files collection for existing data
+        if ($filesTable->hasColumn('private_hash')) {
+            $result= $this->fetchAll('SELECT * FROM directus_files WHERE private_hash is null;');
+            if(count($result) > 0) {
+                foreach($result as $key=>$value) {
+                    $this->execute(\Directus\phinx_update(
+                        $this->getAdapter(),
+                        'directus_files',
+                        [
+                            'private_hash' => get_random_string()
+                        ],
+                        ['id' => $value['id']]
+                    ));
+                }
+            }
+        }
+        // Convert string ID of directus_files into integer
```

Comment:
The current directus_files id is already an auto increment integer. This will override everybody's primary-key, which is what we're trying to prevent with these changes